cli/src/cli_util.rs

```diff
@@ -904,14 +904,19 @@ Set which revision the branch points to with `jj branch set {branch_name} -r <RE
         &self,
         template_text: &str,
     ) -> Result<Box<dyn Template<Commit> + '_>, CommandError> {
-        let language = CommitTemplateLanguage::new(
+        let language = self.commit_template_language()?;
+        self.parse_template(&language, template_text)
+    }
+
+    /// Creates commit template language environment for this workspace.
+    pub fn commit_template_language(&self) -> Result<CommitTemplateLanguage<'_>, CommandError> {
```

Comment:
Suppose the second patch will be squashed, do you want to include this change in your patch, or rebase on top of mine?

Reply:
I'm ready to merge this now. Hopefully my impl is close enough to yours that the conflicts are minimal. :)